File: zuist-cluster-fits/tags/0.2.0/src/main/java/cl/inria/massda/Producer.java


package cl.inria.massda;

import com.rabbitmq.client.Connection;
import com.rabbitmq.client.Channel;
import com.rabbitmq.client.ConnectionFactory;
import java.io.IOException;


class Producer{

	String exchangeName;
	Connection connection;
	Channel channel;

	public Producer(String exchangeName, String hostName, String virtualHost, String userId, String password)  throws IOException  {
		this.exchangeName = exchangeName;
		ConnectionFactory factory = new ConnectionFactory();
		factory.setUsername(userId);
		factory.setPassword(password);
		factory.setVirtualHost(virtualHost);
		factory.setHost(hostName);
		factory.setPort(5672);
		connection = factory.newConnection();
		channel = connection.createChannel();

	}

	public void publish(String message, String routingKey) throws IOException {
		channel.basicPublish(exchangeName, routingKey, null, message.getBytes());
		System.out.println(" [x] sent " + message);
	}

	public void close() throws IOException {
		channel.close();
		connection.close();
	}


}

File: zuist-cluster-fits/trunk/src/main/resources/scripts/wcs/producer.py

#!/usr/bin/env python
import pika
import sys

class Producer(object):
	def __init__(self, exchange_name, host, virtual_host, userid, password):
		"""
		Constructor. Initiate connection with the RabbitMQ server.

		@param exchange_name name of the exchange to send messages to
		@param host RabbitMQ server host 
		@param userid RabbitMQ server username
		@param password RabbitMQ server user's password
		"""
		self.exchange_name = exchange_name
		#self.connection = amqp.Connection(host=host, userid=userid,
		#    password=password, virtual_host='/', insist=False)
		#self.channel = self.connection.channel()
		try:
			credentials = pika.PlainCredentials(userid, password)
			self.connection = pika.BlockingConnection(pika.ConnectionParameters(
	        	host=host, virtual_host=virtual_host, port=5672, credentials=credentials))
			self.channel = self.connection.channel()
		except pika.exceptions.ChannelClosed, e:
			print "ChannelClosed"
		self.channel.exchange_declare(exchange=exchange_name)
		#self.channel.queue_declare(queue='task_queue', durable=True)
		#print "Producer"
		#print self


	def publish(self, message, routing_key):
		"""
		Publish message to exchange using routing key

		@param text message to publish
		@param routing_key message routing key
		"""
		#msg = amqp.Message(message)
		#msg.properties["content_type"] = "text/plain"
		#msg.properties["delivery_mode"] = 2
		#self.channel.basic_publish(exchange=self.exchange_name,
		#                           routing_key=routing_key, msg=msg)

		#channel.queue_declare(queue='task_queue', durable=True)
		self.channel.queue_declare(queue="%ss" % routing_key, durable=True)
		try:
			self.channel.basic_publish(exchange=self.exchange_name,
			                      routing_key=routing_key,
			                      body=message#,
			                      #properties=pika.BasicProperties(
			                      #	content_type = 'application/json'
			                      #   delivery_mode = 2, # make message persistent
			                      )#)
			print " [x] Sent %r" % (message,)
		except pika.exceptions.ChannelClosed, e:
			print "ChannelClosed"
		#print "publish"
		

	def close(self):
		"""
		Close channel and connection
		"""
		#self.channel.queue_delete(queue='')
		self.channel.close()
		self.connection.close()
